src/Containers/Microsoft.NET.Build.Containers/ContainerBuilder.cs

```diff
@@ -57,12 +57,30 @@ public static class ContainerBuilder
         ImageBuilder? imageBuilder;
         if (sourceRegistry is { } registry)
         {
-            imageBuilder = await registry.GetImageManifestAsync(
-                baseImageName,
-                baseImageTag,
-                containerRuntimeIdentifier,
-                ridGraphPath,
-                cancellationToken).ConfigureAwait(false);
+            try
+            {
+                imageBuilder = await registry.GetImageManifestAsync(
+                    baseImageName,
+                    baseImageTag,
+                    containerRuntimeIdentifier,
+                    ridGraphPath,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (RepositoryNotFoundException)
+            {
+                logger.LogError(Resource.FormatString(nameof(Strings.RepositoryNotFound), baseImageName, baseImageTag, registry.RegistryName));
+                return 1;
+            }
+            catch (UnableToAccessRepositoryException)
+            {
+                logger.LogError(Resource.FormatString(nameof(Strings.UnableToAccessRepository), baseImageName, registry.RegistryName));
+                return 1;
+            }
+            catch (ContainerHttpException e)
+            {
+                logger.LogError(e.Message);
```

Comment:
```suggestion
                logger.LogErrorFromException(e);
```

? If you don't think the stack is important, or add `, showStackTrace: true` for more transparency (at the cost of ugliness).

Reply:
Oh wait, is this not guaranteed to be a concrete `TaskLoggerHelper`? Hm. Then maybe consider cloning `LogErrorFromException` to make sure it's actionable.

Reply:
Right, it's not a TLH here. This is also called from the VS codepath, where we already have some issues with error reporting, etc. None of this logging was in the canonical format before this PR, and it also doesn't flow through the VS Publish dialog correctly even if it is in the canonical format.